nifi-nar-bundles/nifi-pgp-bundle/nifi-pgp-processors/src/test/java/org/apache/nifi/processors/pgp/EncryptContentPGPTest.java

```diff
@@ -0,0 +1,293 @@
+                elGamalPrivateKey = secretKey.extractPrivateKey(decryptor);
+                elGamalPublicKey = publicKey;
+            }
+        }
+    }
+
+    @Before
+    public void setRunner() {
+        runner = TestRunners.newTestRunner(new EncryptContentPGP());
+    }
+
+    @Test
+    public void testMissingProperties() {
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testSuccessPasswordBasedEncryptionDefaultProperties() throws IOException, PGPException {
+        runner.setProperty(EncryptContentPGP.PASSPHRASE, PASSPHRASE);
+        runner.enqueue(DATA);
+        runner.run();
+
+        assertSuccess(DEFAULT_SYMMETRIC_KEY_ALGORITHM, PASSPHRASE.toCharArray());
+    }
+
+    @Test
+    public void testSuccessPasswordBasedEncryptionSymmetricKeyAlgorithms() throws IOException, PGPException {
+        for (final SymmetricKeyAlgorithm symmetricKeyAlgorithm : SymmetricKeyAlgorithm.values()) {
+            runner = TestRunners.newTestRunner(new EncryptContentPGP());
+            runner.setProperty(EncryptContentPGP.PASSPHRASE, PASSPHRASE);
+            runner.setProperty(EncryptContentPGP.SYMMETRIC_KEY_ALGORITHM, symmetricKeyAlgorithm.toString());
+            runner.enqueue(DATA);
+            runner.run();
+            assertSuccess(symmetricKeyAlgorithm, PASSPHRASE.toCharArray());
+        }
+    }
+
+    @Test
+    public void testSuccessPasswordBasedEncryptionCompressionAlgorithms() throws IOException, PGPException {
+        for (final CompressionAlgorithm compressionAlgorithm : CompressionAlgorithm.values()) {
+            runner = TestRunners.newTestRunner(new EncryptContentPGP());
+            runner.setProperty(EncryptContentPGP.PASSPHRASE, PASSPHRASE);
+            runner.setProperty(EncryptContentPGP.COMPRESSION_ALGORITHM, compressionAlgorithm.toString());
+            runner.enqueue(DATA);
+            runner.run();
+            assertSuccess(DEFAULT_SYMMETRIC_KEY_ALGORITHM, PASSPHRASE.toCharArray());
+        }
+    }
+
+    @Test
+    public void testSuccessPasswordBasedEncryptionFileEncodingAscii() throws IOException, PGPException {
+        runner.setProperty(EncryptContentPGP.PASSPHRASE, PASSPHRASE);
+        runner.setProperty(EncryptContentPGP.FILE_ENCODING, FileEncoding.ASCII.toString());
+        runner.enqueue(DATA);
+        runner.run();
+        assertSuccess(DEFAULT_SYMMETRIC_KEY_ALGORITHM, PASSPHRASE.toCharArray());
+    }
+
+    @Test
+    public void testSuccessPublicKeyEncryptionRsaPublicKey() throws IOException, InitializationException, PGPException {
+        final PGPPublicKey publicKey = rsaSecretKey.getPublicKey();
+        setPublicKeyService(publicKey);
+        final String publicKeyIdSearch = Long.toHexString(publicKey.getKeyID()).toUpperCase();
+        when(publicKeyService.findPublicKey(eq(publicKeyIdSearch))).thenReturn(Optional.of(publicKey));
+
+        runner.enqueue(DATA);
+        runner.run();
+        assertSuccess(rsaPrivateKey);
+    }
+
```

Comment:
I'd also like to see a unit test here for using both passphrase and public key service.

Reply:
Thanks for the suggestion, will add a new unit test method for this use case.